build.py

```diff
@@ -59,14 +59,20 @@ def get_os():
     reference = "{0}/{1}".format(name, version)
     upload = "https://api.bintray.com/conan/{0}/public-conan".format(username)
 
+    build = os.getenv("BUILD", None)
+    if build:
+        build_env = os.getenv(build, None)
+        if build_env:
+            for env_var in build_env.split():
+                name, value = env_var.split("=", 1)
+                os.environ[name] = value
+
     builder = ConanMultiPackager(
-        username=username,
-        channel=channel,
-        reference=reference,
-        upload=upload,
-        remotes=upload,  # while redundant, this moves bincrafters remote to position 0
-        upload_only_when_stable=True,
```

Comment:
Both `upload_only_when_stable` and `stable_branch_pattern` are useful.

Reply:
Mostly packages are simple and easy to validate. But I agree to upload non-stable packages for Boost and more complex projects.

My vote is keep only stable upload by default, but remove when necessary.

WDYT?

/cc @solvingj @sigmoidal @SSE4